frontend/Store.js

```diff
@@ -614,6 +620,17 @@ class Store extends EventEmitter {
       this._nodesByName = this._nodesByName.set(node.get('name'), this._nodesByName.get(node.get('name')).delete(id));
     }
   }
+
+  _setGetterValue(data: DataType) {
+    var node = this._nodes.get(data.id);
+    var props = node.get('props');
+    var last = data.path.length - 1;
+    var propObj = data.path.slice(0, last).reduce((obj_, attr) => obj_ ? obj_[attr] : null, props);
+    propObj[data.path[last]] = data.value;
+    var newProps = assign({}, props);
```

Comment:
Reason why - you don't mutate original props on requesting getter value. If you edit some prop(for example text) in devtools, it forwards changes to original page component. Original component props mutates and new set of props goes back to devtools store, immutable collection recreates, UI rerenders etc

But on requesting getter value we pass just singular value and do not make any changes to original prop. So this part of code needed to make frontend store's data in sync and refresh UI nicely.

But why do we need this line:
`var newProps = assign({}, props);`

Data stored in immutable map on store._nodes But carried props is not immutable map, it just a plain JS object. In Node.js (it's component node on left panel) where is  SCU check

`  shouldComponentUpdate(nextProps: PropsType) {
    return nextProps !== this.props;
  }`

So if we just swap value inside this object UI wont refreshes correclty